File: contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        //check for null input 
        if(nums == null || nums.length ==0)return false;
        HashSet<Integer> numSet = new HashSet<>();
        for(int num: nums){
            if(numSet.contains(num)) return true;
            numSet.add(num);
        }
        return false;
    }
}
//time and space complexity: O(n)

File: contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        #check for null input
        if nums is None: return False
        numSet = set()
        for num in nums:
            if num in numSet: return True
            numSet.add(num)
        return False
    #time and space complexity:O(n)